### ISSUE TITLE
Don't forcibly resign the Tsar if the resignation will cause the game to be paused

### ISSUE THREAD

**Issue by zuzak** [grifferz/pah-irc#98]
If the game is going to be paused anyway, we may as well indefinitely extend the "decision round" and give the Tsar longer to decide, so that we don't waste the hand.

Vaguely related to #95.

**Comment by grifferz** [comment]
But players can't join once the game is waiting for the Tsar to pick a winner, so this would potentially stall the game forever if the Tsar never returns.

**Comment by zuzak** [comment]
Blocked by #39.
On 13 Mar 2015 19:51, "grifferz" notifications@github.com wrote:

But players can't join once the game is waiting for the Tsar to pick a
winner, so this would potentially stall the game forever if the Tsar never
returns.

**Comment by grifferz** [comment]
But #39 doesn't help either because all that asks for is that the new player be dealt in to the next hand. If the Tsar is never to be resigned then how do you deal with a Tsar that never returns?

**Comment by zuzak** [comment]
The game is de facto paused anyway, so it doesn't really matter.

If a new player joins, and the turn clock has not elapsed, continue
waiting. If a new player joins and the turn clock has expired, resign the
Tsar. If the turn clock elapses and there are no new players, continue
waiting until one does.

**Comment by grifferz** [comment]
Doesn't that result in exactly the same outcome, but more complicated?

### Scenario 1: New player will join before turnclock expires
1. Hand is completed but Tsar is absent.
2. New player tries to join but can't because hand is complete.
3. Tsar is resigned, hand is a bust, no one wins, game is paused.
4. New player joins and game resumes.

### Scenario 2: New player will join after turnclock expires
1. Hand is completed but Tsar is absent.
2. Tsar is resigned, hand is a bust, no one wins, game is paused.
3. New player joins and game resumes.

### Scenario 3: Like (1) but with this proposal implemented
1. Hand is completed but Tsar is absent.
2. New player tries to join and is told they will be dealt into the next hand.
3. Tsar is resigned, hand is a bust, no one wins, game proceeds with new player.

### Scenario 4: Like (2) but with this proposal implemented
1. Hand is completed but Tsar is absent.
2. Tsar would normally be resigned but because this would pause the game instead they are given longer.
3. New player joins and is told they'll be dealt in to next hand.
4. Tsar is resigned, hand is a bust, no one wins, game proceeds with new player.

It seems to me that in all cases the hand is a bust and waits for a new player to get it going again, and the only useful difference is automatically dealing a new player in to the next hand, which is issue #39.

**Comment by zuzak** [comment]
Yes, but with this proposal the Tsar is hopefully more likely to respond in
time (because the time has been extended), and won't bust the round.

**Comment by grifferz** [comment]
Hmm, OK, I see your point. This does only seem to be an issue for the smaller game in #debug though. Perhaps we will see it as more of an issue when/if people start to get bored of the game.
